src/parser/type_checking.c

```diff
@@ -22979,11 +22978,19 @@ pt_common_collation (PT_COLL_INFER * arg1_coll_infer, PT_COLL_INFER * arg2_coll_
 	{
 	  goto error;
 	}
-      *common_coll = arg2_coll_infer->coll_id;
-      *common_cs = arg2_coll_infer->codeset;
+      if (arg2_coll_infer->coll_id < arg1_coll_infer->coll_id)
+	{
+	  *common_coll = arg1_coll_infer->coll_id;
+	  *common_cs = arg1_coll_infer->codeset;
+	}
+      else
+	{
+	  *common_coll = arg2_coll_infer->coll_id;
+	  *common_cs = arg2_coll_infer->codeset;
+	}
```

Comment:
coll_id has no meaning to compare. it is just an index of lang_Collations array.

Reply:
ok, I fixed it.